### PR TITLE
[IOTDB-5491] Add IoTConsumus memory control metric items

### PR DESCRIPTION
**Pull request by BUAAserein**
#### Add queue memory and synchronization memory items to distinguish memory occupation

Please see details in [IOTDB-5491](https://issues.apache.org/jira/browse/IOTDB-5491).

### CODE REVIEW

File: consensus/src/main/java/org/apache/iotdb/consensus/iot/logdispatcher/IoTConsensusMemoryManagerMetrics.java [hunk 1]
```diff
@@ -42,11 +42,29 @@ public void bindTo(AbstractMetricService metricService) {
         IoTConsensusMemoryManager::getMemorySizeInByte,
         Tag.NAME.toString(),
         "IoTConsensus");
+    metricService.createAutoGauge(
+        Metric.MEM.toString(),
+        MetricLevel.IMPORTANT,
+        iotConsensusMemoryManager,
+        IoTConsensusMemoryManager::getQueueMemorySizeInByte,
+        Tag.NAME.toString(),
+        "IoTConsensus_Queue");
```
Comment (OneSizeFitsQuorum) [inline consensus/src/main/java/org/apache/iotdb/consensus/iot/logdispatcher/IoTConsensusMemoryManagerMetrics.java:51]: Remove `-` as you have used Camel-Case naming schema.

File: consensus/src/main/java/org/apache/iotdb/consensus/iot/logdispatcher/IoTConsensusMemoryManagerMetrics.java [hunk 2]
```diff
@@ -42,11 +42,29 @@ public void bindTo(AbstractMetricService metricService) {
...
+    metricService.createAutoGauge(
+        Metric.MEM.toString(),
+        MetricLevel.IMPORTANT,
+        iotConsensusMemoryManager,
+        IoTConsensusMemoryManager::getSyncMemorySizeInByte,
+        Tag.NAME.toString(),
+        "IoTConsensus_Sync");
```
Comment (OneSizeFitsQuorum) [inline consensus/src/main/java/org/apache/iotdb/consensus/iot/logdispatcher/IoTConsensusMemoryManagerMetrics.java:58]: same

**Review by SpriCoder — CHANGES_REQUESTED** [review]
Excellent Work!
Just one problem, Please update the en and zh doc at same time
en: docs/UserGuide/Monitor-Alert/Metric-Tool.md
zh: docs/zh/UserGuide/Monitor-Alert/Metric-Tool.md

File: consensus/src/main/java/org/apache/iotdb/consensus/iot/logdispatcher/IoTConsensusMemoryManagerMetrics.java (hunk 1 above)
Comment (BUAAserein) [reply consensus/src/main/java/org/apache/iotdb/consensus/iot/logdispatcher/IoTConsensusMemoryManagerMetrics.java]: ok

File: consensus/src/main/java/org/apache/iotdb/consensus/iot/logdispatcher/IoTConsensusMemoryManagerMetrics.java (hunk 2 above)
Comment (BUAAserein) [reply consensus/src/main/java/org/apache/iotdb/consensus/iot/logdispatcher/IoTConsensusMemoryManagerMetrics.java]: ok

**Comment by BUAAserein** [comment]
> Excellent Work! Just one problem, Please update the en and zh doc at same time en: docs/UserGuide/Monitor-Alert/Metric-Tool.md zh: docs/zh/UserGuide/Monitor-Alert/Metric-Tool.md

ok

**Review by SpriCoder — APPROVED** [review]
LGTM~

**Review by OneSizeFitsQuorum — APPROVED** [review]
LGTM

**Pull request merged**